aspnetcore/blazor/fundamentals/signalr.md

```diff
@@ -251,6 +251,161 @@ There was a problem with the connection! (Current reconnect attempt: 3 / 8)
 
 By default, Blazor Server apps prerender the UI on the server before the client connection to the server is established. For more information, see <xref:mvc/views/tag-helpers/builtin-th/component-tag-helper>.
 
+:::moniker-end
+
+:::moniker range=">= aspnetcore-8.0"
+
+## Monitor circuit activity (Blazor Server)
+
+Monitor inbound circuit activity in Blazor Server apps using the `CreateInboundActivityHandler` method on <xref:Microsoft.AspNetCore.Components.Server.Circuits.CircuitHandler>. Inbound circuit activity is any activity sent from the browser to the server, such as UI events or JavaScript-to-.NET interop calls.
+
+For example, you can use a circuit activity handler to detect if the client is idle:
+
+```csharp
+public sealed class IdleCircuitHandler : CircuitHandler, IDisposable
+{
+    readonly Timer timer;
+    readonly ILogger logger;
+
+    public IdleCircuitHandler(IOptions<IdleCircuitOptions> options, 
+        ILogger<IdleCircuitHandler> logger)
+    {
+        timer = new Timer();
+        timer.Interval = options.Value.IdleTimeout.TotalMilliseconds;
+        timer.AutoReset = false;
+        timer.Elapsed += CircuitIdle;
+        this.logger = logger;
+    }
+
+    private void CircuitIdle(object? sender, System.Timers.ElapsedEventArgs e)
+    {
+        logger.LogInformation(nameof(CircuitIdle));
+    }
+
+    public override Func<CircuitInboundActivityContext, Task> CreateInboundActivityHandler(
+        Func<CircuitInboundActivityContext, Task> next)
+    {
+        return context =>
+        {
+            timer.Stop();
+            timer.Start();
+            return next(context);
+        };
+    }
+
+    public void Dispose()
+    {
+        timer.Dispose();
+    }
+}
+
+public class IdleCircuitOptions
+{
+    public TimeSpan IdleTimeout { get; set; } = TimeSpan.FromMinutes(5);
+}
+
+public static class IdleCircuitHandlerServiceCollectionExtensions
+{
+    public static IServiceCollection AddIdleCircuitHandler(
+        this IServiceCollection services, 
+        Action<IdleCircuitOptions> configureOptions)
+    {
+        services.Configure(configureOptions);
+        services.AddIdleCircuitHandler();
+        return services;
+    }
+
+    public static IServiceCollection AddIdleCircuitHandler(
+        this IServiceCollection services)
+    {
+        services.AddScoped<CircuitHandler, IdleCircuitHandler>();
+        return services;
+    }
+}
+```
+
+Circuit activity handlers also provide a way to access scoped Blazor services from other non-Blazor dependency injection (DI) scopes, such as scopes created using <xref:System.Net.Http.IHttpClientFactory>. There's an existing pattern for accessing circuit-scoped services from other dependency injection scopes, but it requires using a custom base component type. With circuit activity handlers, we can use a better approach, which is demonstrated by the following example:
```

Comment:
> There's an existing pattern for accessing circuit-scoped services from other dependency injection scopes...

The docs for this pattern are [here](https://learn.microsoft.com/aspnet/core/blazor/fundamentals/dependency-injection?view=aspnetcore-7.0#access-blazor-services-from-a-different-di-scope). Given that we now have a better way of doing this, maybe we could modify the 8.0 version of those docs to point to this section?

Reply:
And we could probably go as far as removing that example entirely from the 8.0 docs